src/main/java/com/alipay/oceanbase/rpc/location/LocationUtil.java

```diff
@@ -945,11 +948,18 @@ private static ObPartitionEntry getPartitionLocationFromResultSet(TableEntry tab
                                                                                    throws SQLException,
                                                                                    ObTablePartitionLocationRefreshException {
         Map<Long, ObPartitionLocation> partitionLocation = new HashMap<Long, ObPartitionLocation>();
+        Map<Long, Long> tabletLsIdMap = new HashMap<>();
         while (rs.next()) {
             ReplicaLocation replica = buildReplicaLocation(rs);
             long partitionId;
             if (ObGlobal.obVsnMajor() >= 4) {
                 partitionId = rs.getLong("tablet_id");
+                long lsId = rs.getLong("ls_id");
+                if (!rs.wasNull()) {
+                    tabletLsIdMap.put(partitionId, lsId);
+                } else {
+                    tabletLsIdMap.put(partitionId, INVALID_LS_ID); // non-partitioned table
```

Comment:
If you put your tablet-ls map in partitionEntry, it is ok to put INVALID_LS_ID
Since the tablet-ls map is in partitionEntry(contains location info), which is different from other tablet/partition info, I guess we should leave a comment in partitionEntry to declare that other info about the tablet could be found on partitionInfo. On the other hand, comment on partitionInfo that the tablet-ls map is in partitionEntry.